lib/types/entry.ts

```diff
@@ -42,109 +42,88 @@ export type BasicEntryField =
   | EntryFields.RichText
   | EntryFields.Object
 
+type BaseEntry = {
+  sys: EntrySys
+  metadata: Metadata
+}
+
 /**
  * @category Entities
  */
-export interface Entry<T> {
-  sys: EntrySys
-  metadata: Metadata
-  fields: T
+export type GenericEntry<Fields extends FieldsType> = BaseEntry & {
+  fields: Fields
 }
 
-export interface EntryWithAllLocalesAndWithoutLinkResolution<
+/**
+ * @category Entities
+ * @deprecated
+ */
+export type Entry<Fields extends FieldsType> = GenericEntry<Fields>
+
+export type ResolvedField<
   Fields extends FieldsType,
-  Locales extends LocaleCode
-> {
-  sys: EntrySys
-  metadata: Metadata
-  fields: {
-    [FieldName in keyof Fields]: {
-      [LocaleName in Locales]?: Fields[FieldName] extends EntryFields.Link<any>
-        ? EntryLink
-        : Fields[FieldName] extends EntryFields.Link<any>[]
-        ? EntryLink[]
-        : Fields[FieldName]
-    }
-  }
+  Modifiers extends ChainModifiers,
+  Locales extends LocaleCode = any
+> = Fields extends EntryFields.Link<infer LinkedFields>
+  ? 'WITHOUT_LINK_RESOLUTION' extends Modifiers
+    ? EntryLink
+    :
+        | NewEntry<LinkedFields, Modifiers, Locales>
+        | ('WITHOUT_UNRESOLVABLE_LINKS' extends Modifiers ? undefined : EntryLink)
+  : Fields extends EntryFields.Link<infer LinkedFields>[]
+  ? 'WITHOUT_LINK_RESOLUTION' extends Modifiers
```

Comment:
This check is similar to the one for single reference field and we could introduce a separate type that only has this check but then we’d have the check for each array element instead of once for the whole list and that didn’t feel cleaner. I’m open to suggestions though.

Reply:
I have a solution but it requires changing more than one type so it’ll be in the next PR.